### PR TITLE
CI: Add the shimv2 testing cases

### PR DESCRIPTION
**Pull request by lifupan**
Add the containerd-shim-kata-v2 testing cases
based on cri+containerd. 

Fixes: #616

Signed-off-by: Fupan Li <lifupan@gmail.com>

### CODE REVIEW

**Comment by chavafg** [comment]
can you add an space after the colon? 
`Depends-on: github.com/kata-containers/runtime#572`

and thanks for raising the issue, needs to be resolved.

**Comment by sboeuf** [comment]
@lifupan could you rebase and fix this PR? From my understanding, the test you introduced here should not impact the current tests if containerd-shim-v2 is not implemented, right?
If so, we should be able to merge it before we move forward with the implementation on the runtime side.

**Comment by raravena80** [comment]
@lifupan ping (from your weekly Kata herder)

**Comment by jodh-intel** [comment]
From a CI log:

```
This PR depends on repository: https and pull request:  https
package https: unrecognized import path "https" (import path does not begin with hostname)
```

Ping @kata-containers/tests - we've landed the shimv2 code so we really need to get this landed so it's being tested!!

**Comment by chavafg** [comment]
@lifupan can you amend your commit replacing the Depends-on tag with:
```
Depends-on: github.com/kata-containers/runtime#940
```

**Comment by lifupan** [comment]
Ping @kata-containers/tests

**Comment by gnawux** [comment]
LGTM

**Comment by gnawux** [comment]
@jodh-intel I think this is close to being merged now.

File: integration/containerd/cri/integration-tests.sh [hunk 1]
```diff
@@ -60,10 +74,22 @@ ci_cleanup() {
 	if [ "$ID" == ubuntu ] &&  [ -n "${CI}" ] ;then
 		[ -f "${kata_config}" ] && sudo rm "${kata_config}"
 	fi
+
+	if [ -n "${FACTORY_TEST}" ]; then
+		echo "destroy vm template"
+	        sudo -E PATH=$PATH "$RUNTIME" factory destroy
+	fi
+
```
Comment (jodh-intel) [inline integration/containerd/cri/integration-tests.sh]: Should this function "reset" the changes made in `ci_config()`? In other words, undo:

- `enable_template = true`
- `internetworking_model = "bridged"`

**Comment by jodh-intel** [comment]
I just have one question. If the cleanup isn't required then...

lgtm

File: integration/containerd/cri/integration-tests.sh (hunk 1 above)
Comment (lifupan) [reply integration/containerd/cri/integration-tests.sh]: Hi @jodh-intel 
The ci_cleanup() will "reset" the changes by  remove the config file.

Comment (jodh-intel) [reply integration/containerd/cri/integration-tests.sh]: Great! ;)

**Comment by jodh-intel** [comment]
lgtm

**Pull request merged**